pkg/image/proxy_types.go

```diff
@@ -0,0 +1,122 @@
+/*
+Copyright 2017 Google, Inc. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package image
+
+import (
+	"context"
+	"io"
+
+	"github.com/containers/image/docker/reference"
+
+	"github.com/containers/image/types"
+	digest "github.com/opencontainers/go-digest"
+)
+
+// ProxySource is a type that implements types.ImageSource by proxying all calls to an underlying implementation.
+type ProxySource struct {
+	Ref types.ImageReference
```

Comment:
You could consider embedding these three types

Reply:
Done. Left the one un-embedded.